### PR TITLE
Updated C28726

### PR DESCRIPTION
**Pull request by MugBergerFries**
Added more information, an example, and a list of the banned functions covered by this rule

### CODE REVIEW

**Review by Rastaban — APPROVED** [review]

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md [hunk 1]
```diff
@@ -9,11 +9,56 @@ f1_keywords:
 # C28726
 
 
-warning C28726: Banned API Usage
+**warning C28726: Banned API Usage L2**\
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md:12]:
```suggestion
**Warning C28726: Banned API Usage L2**\
```

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md [hunk 2]
```diff
@@ -9,11 +9,56 @@ f1_keywords:
...
+_Banned API Usage:  *function name* is a Banned API as listed in dontuse.h for security purposes._
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md:13]: Is `dontuse.h` a public header? If not, we should remove this reference.

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md [hunk 3]
```diff
@@ -9,11 +9,56 @@ f1_keywords:
...
-This warning indicates that a function is being used that has been banned or deprecated, and has a more robust and secure replacement.
-
-To find the replacement, search for the name of the deprecated function in the [Windows Driver Development](../index.yml) documentation. The documentation lists the name of the deprecated function and suggests replacements.
-
- 
+This warning indicates that a function is being used that has been banned and has a more robust or secure replacement. This specific error indicates that the banned function has potential to overflow a buffer. A list of all banned functions covered by this error, why they are banned, and recommended replacements can be found after the following example: 
+ ## Example 
 
+The following code generates this warning: 
+```
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md:19]: I think you can get markdown to do syntax highlighting by specifying the language here.

Something like this maybe?
```suggestion
```c
```

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md [hunk 4]
```diff
@@ -9,11 +9,56 @@ f1_keywords:
...
+void example_func() 
+{ 
+    char user_input[10]; 
+    scanf(“%s”, input); // scanf is banned for security purposes 
+} 
+```
+This is due to the use of the unsafe function scanf. scanf does place any limit on the size of the data copied to the buffer. To fix this issue, we can use scanf_s, the safe replacement to this function. scanf_s requires the developer to specify how many bytes are intended to be copied. scanf_s will ensure only that many bytes are copied. For example, the following code is safe: 
+```
```
Comment (MichaelSquires) [inline windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md:27]: Enable syntax highlighting if you were able to get it working above.

**Review by MichaelSquires — CHANGES_REQUESTED** [review]

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md (hunk 3 above)
Comment (MugBergerFries) [reply windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md]: Added cpp to this PR, will add to others now!

File: windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md (hunk 2 above)
Comment (Rastaban) [reply windows-driver-docs-pr/devtest/28726-banned-api-usage-use-updated-function-replacement.md]: I had a conversation offline about this.  The file dontuse.h is public (it is in the windows SDK) so it is okay being referenced.  However, I don't think it is helpful to reference the file.  I am creating a PR on the tool to update the message, after that goes through the message here can be updated to match.

**Review by MichaelSquires — APPROVED** [review]

**Comment by MugBergerFries** [comment]
@DOMARS This is approved and ready to be merged

**Pull request merged**